data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManagerTests.java

```diff
@@ -238,6 +238,25 @@ public void getIndexName_DoesRetryOnOpenSearchExceptions_UntilFailure() throws I
         verify(innerIndexManager, times(3)).setupIndex();
     }
 
+    @Test
+    public void getIndexName_IndexWithDateTimePattern_NotAsSuffix() throws IOException {
+        final String dynamicIndexName = "test-index-%{yyyy.MM.dd}-randomtext";
+        final String indexWithDateTimePatternResolved = "test-index-2023.11.11-randomtext";
+        when(indexConfiguration.isNormalizeIndex()).thenReturn(true);
+        innerIndexManager = mock(IndexManager.class);
+
+        when(mockIndexManagerFactory.getIndexManager(
+                IndexType.CUSTOM, openSearchClient, restHighLevelClient, openSearchSinkConfiguration, templateStrategy, indexWithDateTimePatternResolved)).thenReturn(innerIndexManager);
+        when(innerIndexManager.getIndexName(indexWithDateTimePatternResolved)).thenReturn(indexWithDateTimePatternResolved);
+
+        try (final MockedStatic<AbstractIndexManager> abstractIndexManagerMockedStatic = mockStatic(AbstractIndexManager.class)) {
+            abstractIndexManagerMockedStatic.when(() -> AbstractIndexManager.getIndexAliasWithDate(dynamicIndexName))
+                    .thenReturn(indexWithDateTimePatternResolved);
+            final String result = dynamicIndexManager.getIndexName(dynamicIndexName);
+            assertThat(result, equalTo(indexWithDateTimePatternResolved));
```

Comment:
Please move this assertion out of the try block:

```
try (final MockedStatic<AbstractIndexManager> abstractIndexManagerMockedStatic = mockStatic(AbstractIndexManager.class)) {
            abstractIndexManagerMockedStatic.when(() -> AbstractIndexManager.getIndexAliasWithDate(dynamicIndexName))
                    .thenReturn(indexWithDateTimePatternResolved);
            
}
final String result = dynamicIndexManager.getIndexName(dynamicIndexName);
assertThat(result, equalTo(indexWithDateTimePatternResolved));
```

Reply:
I cannot get the assertion work outside of try-block likely because the static mock is scoped in the try-block.